### PR TITLE
Fix regex matching tags and improvements.

### PR DESCRIPTION
**Pull request by nicolafiorillo**
I have found a regex bug matching tags: not indented lines were not correctly matched.
I took the opportunity to improve the same regex with more flexible patterns (spaces between # and tag, spaces between tag and the rest of the line).

Thanks.

### CODE REVIEW

File: lib/credo/check/design/tag_helper.ex [hunk 1]
```diff
@@ -4,7 +4,7 @@ defmodule Credo.Check.Design.TagHelper do
   alias Credo.Check.CodeHelper
 
   def tags(source, tag_name) do
-    {:ok, regex} = Regex.compile("[^\?]# #{tag_name}:? .+", "i")
+    {:ok, regex} = Regex.compile("#\s*#{tag_name}:?\s*.+", "i")
```
Comment (rrrene) [inline lib/credo/check/design/tag_helper.ex]: Why did you remove the leading `[^\?]`? It was meant as a safeguard against mistaking `?#` for a comment ...

Comment (nicolafiorillo) [inline lib/credo/check/design/tag_helper.ex]: Match on (possible) `?#` restored.
But I did not get the point: why just `?#` and not any other char?
And writing ?# is not a syntax error, catch by compiler before credo? Trying to make a test about this, credo exits with a `** (RuntimeError) Source could not be parsed!`

Comment (rrrene) [inline lib/credo/check/design/tag_helper.ex]: `?` can be used to describe a character (`?#` is equal to writing `'#'`, `?A` is equal to writing `'A'`).

We just want to make sure that the `#` we match is acutally indicating a comment.
